src/main/java/org/opensearch/flowframework/util/ApiSpecFetcher.java

```diff
@@ -81,7 +81,7 @@ public static boolean compareRequiredFields(List<String> requiredEnumParams, Str
 
             List<String> requiredApiParams = schema.getRequired();
             if (requiredApiParams != null && !requiredApiParams.isEmpty()) {
-                return new HashSet<>(requiredEnumParams).equals(new HashSet<>(requiredApiParams));
+                return new HashSet<>(requiredEnumParams).containsAll(new HashSet<>(requiredApiParams));
```

Comment:
I don't think you need to convert the ApiParams list to a set; containsAll takes a collection.